.github/workflows/ci.yml

```diff
@@ -49,6 +49,14 @@ jobs:
       - name: Run tests (pattern matching)
         if: matrix.python == '3.10'
         run: pytest tests/test_pattern_matching.py
+
+      # Linters
+      - name: Run flake8 (Python >= 3.10)
+        run: flake8
+        if: matrix.python != '3.9' && matrix.python != '3.8' && matrix.python != '3.7'
+      - name: Run flake8 (Python < 3.10)
+        run: flake8 --extend-exclude tests/test_pattern_matching.py
+        if: matrix.python == '3.9' || matrix.python == '3.8' || matrix.python == '3.7'
```

Comment:
This seems reasonable for now.
I don't use GitHub Actions myself either. 
We can improve it in the future if we find a better approach.

Reply:
Only this I could suggest is, since Pattern matching will be in all version >= 3.10, we could do a `>` comparison... but the versions are strings, so that probably wouldn't even work here?

Reply:
yeah i wondered about the same thing. it's a string indeed, and while string comparison likely works it would sort `3.10` before `3.9`. it seems there is no version comparison operator:

https://docs.github.com/en/actions/learn-github-actions/expressions